pkg/engine/engine.go

```diff
@@ -905,7 +905,15 @@ func (e *Engine) verificationOverlapWorker(ctx context.Context) {
 					detectorKeysWithResults[detector.Key] = detector
 				}
 
-				results = e.filterResults(ctx, detector, results)
+				// If results filtration eliminates a rotated secret, then that rotation will never be reported. This
+				// problem can theoretically occur for any scan, but we've only actually seen it in practice during
+				// targeted scans. (The reason for this discrepancy is unclear.) The simplest fix is therefore to
+				// disable filtration for targeted scans, but if you're here because this problem surfaced for a
+				// non-targeted scan then we'll have to solve it correctly.
+				if chunk.chunk.SecretID == 0 {
```

Comment:
I know there are comments saying `SecretID != 0` for targeted scans, but what do you think about adding a little helper function to whatever type `chunk.chunk` is.

```go
func (c Chunk) TargetedScan() bool { return c.SecretID != 0 }
```

Reply:
I like the sentiment, but I also hate that whether a scan is targeted is stored in _chunk_ data and I think I'd prefer to not double down on that.